### PR TITLE
Testing clumping

### PR DESCRIPTION
**Pull request by lizvance**

### CODE REVIEW

File: static/downloadables/grep_file.py [hunk 1]
```diff
@@ -142,7 +147,8 @@ def formatVarForFiltering(traits, studyTypes, studyIDs, ethnicities, sexes, valu
 
     return traits, studyTypes, studyIDs, ethnicities, sexes, valueTypes
 
-def filterTXT(tableObjDict, allClumpsObjDict, studySnpsDict, inputFiles, filteredFilePath, traits, studyIDs, studyTypes, ethnicities, valueTypes, sexes, isAllFiltersNone, p_cutOff):
+def filterTXT(tableObjDict, allClumpsObjDict, studySnpsDict, inputFiles, filteredFilePath, traits, studyIDs, studyTypes, ethnicities, valueTypes, sexes, isAllFiltersNone, p_cutOff, useGWASupload):
+    # txt_file = openFileForParsing(inputFiles)
```
Comment (mpage21) [inline static/downloadables/grep_file.py:151]: We can remove this line

File: static/downloadables/grep_file.py [hunk 2]
```diff
@@ -263,8 +269,8 @@ def filterVCF(tableObjDict, allClumpsObjDict, studySnpsDict, inputFiles, filtere
             
             firstFile = False
 
-            if fileEmpty:
-                raise SystemExit("The VCF file is either empty or formatted incorrectly. Each line must have 'GT' (genotype) formatting and a non-Null value for the chromosome and position")
+        if fileEmpty:
```
Comment (mpage21) [inline static/downloadables/grep_file.py:272]: Need to check this out again. (note for myself)

File: static/downloadables/grep_file.py [hunk 3]
```diff
@@ -348,7 +354,10 @@ def shouldUseAssociation(traits, studyIDs, studyTypes, ethnicities, sexes, value
     useValueType = False
     useSex = False
 
-    studyValueTypes = set([x.split("|")[-1] for x in studyMetaData['traits'][trait]['pValBetaAnnoValType']])
+    if studyMetaData is not None:
```
Comment (mpage21) [inline static/downloadables/grep_file.py:357]: Why this change?

File: static/downloadables/parse_associations.py [hunk 4]
```diff
@@ -52,23 +53,27 @@ def getDownloadedFiles(fileHash, requiredParamsHash, superPop, mafCohort, refGen
     specificAssociPath = os.path.join(basePath, "associations_{ahash}.txt".format(ahash = fileHash))
     # get the paths for the associationsFile and clumpsFile
     if useGWASupload:
+        print('in use gwas upload')
```
Comment (mpage21) [inline static/downloadables/parse_associations.py:56]: Do we want to keep this here?

File: static/downloadables/parse_associations.py [hunk 5]
```diff
@@ -238,7 +248,7 @@ def parse_txt(filteredFilePath, clumpsObjDict, tableObjDict, snpSet, clumpNumDic
                             # The snp wasn't in the clump map (meaning it wasn't in 1000 Genomes), so add it
                             else:
                                 sample_map[snp] = alleles
-                        else:
+                        elif riskAllele not in alleles and isIndividualClump:
```
Comment (mpage21) [inline static/downloadables/parse_associations.py:251]: Does this cover all the cases? I'm just worried if we will be missing something by not having an 'else'

File: static/downloadables/parse_associations.py [hunk 6]
```diff
@@ -451,6 +463,7 @@ def parse_vcf(filteredFilePath, clumpsObjDict, tableObjDict, snpSet, clumpNumDic
                                                     # and switch out the data accordingly
                                                     if pValue < index_pvalue:
                                                         index_snp_map[sample][clumpNum] = rsID, riskAllele, alleles if complements is None else complements
+
```
Comment (mpage21) [inline static/downloadables/parse_associations.py:466]: space can be removed

File: static/downloadables/grep_file.py (hunk 3 above)
Comment (lizvance) [reply static/downloadables/grep_file.py:357]: I was running into some cases where the studyMetaData was None and so the studyValueTypes initiation was throwing an error.

File: static/downloadables/parse_associations.py (hunk 4 above)
Comment (lizvance) [reply static/downloadables/parse_associations.py]: Nope :)

File: static/downloadables/parse_associations.py (hunk 5 above)
Comment (lizvance) [reply static/downloadables/parse_associations.py:250]: I'm pretty sure it covers all cases, but I'll change it back to 'else' just for cleanliness. I was writing it out more just so I could see the logic myself.

File: static/downloadables/grep_file.py (hunk 3 above)
Comment (mpage21) [reply static/downloadables/grep_file.py:356]: When was it None? if it is none, then we most likely have a problem

Comment (lizvance) [reply static/downloadables/grep_file.py:357]: Oh, I wrote that code before I put in the check to make sure users weren't specifying any study filters if they were also uploading their own GWAS data. We system exit with an explanation now, but before we were setting the studyMetaData to None if the studyID wasn't in the database and then still trying to get the studyBalueTypes from it. I'm going to remove the if statement, though, because that will no longer happen.

File: static/downloadables/calculate_score.py [hunk 7]
```diff
@@ -215,7 +213,10 @@ def vcfcalculations(snpSet, vcfObj, tableObjDict, mafDict, isJson, isCondensedFo
                         'pValueAnnotation': pValueAnno,
                         'betaAnnotation' : betaAnnotation,
                         'scoreType': valueType,
-                        'units (if applicable)': studyUnits
+                        'units (if applicable)': studyUnits,
+                        'snpOverlap': snpOverlap,
+                        'totalSnps': totalSnps,
+                        'UsedSuperPop': preferredPop
```
Comment (lizvance) [inline static/downloadables/calculate_score.py:219]: I'll change this to a lowercase u

File: static/downloadables/calculate_score.py [hunk 8]
```diff
@@ -276,20 +277,25 @@ def formatJson(studyInfo, outputFile):
...
+    valueType = valueType.lower()
 # calculate the PRS from the list of betas
     ploidy = 2
     combinedBetas = 0
     # if there are no values in the betas array, set combinedBetas to 'NF'
     if not betas:
         combinedBetas = "NF"
+    elif nonMissingSnps == 0:
+        combinedBetas = "NF"
+        SystemExit(f"ERROR: We had values that were zero, so the snps weren't added to the lists. Please let us know about this error and send up this studyID: {studyID}")
```
Comment (lizvance) [inline static/downloadables/calculate_score.py:297]: This error is a little confusing for me...so i imagine a user will be really confused. What values were zero? What lists were the snps not added to?

File: static/downloadables/connect_to_server.py [hunk 9]
```diff
@@ -668,6 +713,11 @@ def getVariantAlleles(rsID, mv):
     output = f.getvalue()
 
     objs = queryResult['hits'][0] if len(queryResult['hits']) > 0 else None
+    #TODO if there is only one item in the objs list, the type is not a list, it's just a map
```
Comment (mpage21) [inline static/downloadables/connect_to_server.py:716]: Is this TODO finished?

File: static/downloadables/connect_to_server.py [hunk 10]
```diff
@@ -815,32 +865,31 @@ def checkInternetConnection():
...
         filteredKeys = []
         superPopHeirarchy = {
             'EUR': ['EUR', 'AMR', 'SAS', 'EAS', 'AFR'],
             'AMR': ['AMR', 'EUR', 'SAS', 'EAS', 'AFR'],
-            'SAS': ['SAS', 'EAS', 'EUR', 'AMR', 'AFR'],
-            'EAS': ['EAS', 'SAS', 'EUR', 'AMR', 'AFR'],
+            'SAS': ['SAS', 'EAS', 'AMR', 'EUR', 'AFR'],
+            'EAS': ['EAS', 'SAS', 'AMR', 'EUR', 'AFR'],
             #TODO: check with justin if these heirarchies are correct
-            'AFR': ['AFR', 'EUR', 'AMR', 'SAS', 'EAS']
```
Comment (mpage21) [inline static/downloadables/connect_to_server.py:830]: Can we remove this TODO?

File: static/downloadables/runPrsCLI.sh [hunk 11]
```diff
@@ -661,9 +661,10 @@ calculatePRS () {
                     echo -e "${LIGHTRED}Quitting...${NC}"
                     exit 1
                 fi
-                output=$(echo $OPTARG | tr '[:upper:]' '[:lower:]')
+                output=$(echo $OPTARG)
```
Comment (mpage21) [inline static/downloadables/runPrsCLI.sh:664]: The TODO on line 499 needs to be done

File: static/downloadables/connect_to_server.py (hunk 10 above)
Comment (lizvance) [reply static/downloadables/connect_to_server.py:830]: yes, i'll do that

File: static/downloadables/connect_to_server.py (hunk 9 above)
Comment (lizvance) [reply static/downloadables/connect_to_server.py]: yep!

**Review by MattCloward — APPROVED** [review]
Looks great!
